File: leet/palindrome-number.ts
/**
 * https://leetcode.com/problems/palindrome-number/description/
 * Given an integer x, return true if x is a palindrome, and false otherwise.
 */
export function isPalindrome(x: number): boolean {
  if (x < 0) return false;
  const s = x.toString();
  let start = 0;
  let end = s.length - 1;
  while (start < end) {
    if (s[start] !== s[end]) return false;
    start++;
    end--;
  }
  return true;
}

File: returned/python/palindrome_number.py
# https://leetcode.com/problems/palindrome-number/

class Solution:
    def isPalindrome(self, x: int) -> bool:
        y = str(x)
        l, r = 0, len(y) - 1
        while l < r:
            if y[l] != y[r]:
                return False
            l += 1
            r -= 1
        return True
